backend/open_webui/utils/oauth.py

```
@@ -238,206 +240,336 @@ async def login(self, request, provider):
-                raise HTTPException(400, detail=ERROR_MESSAGES.INVALID_CRED)
-
-            # Check if the user exists
-            user = Users.get_user_by_oauth_sub(provider_sub)
+                    log.error(f"Failed to get token for {provider_name}: {str(e)}")
+                    return None
+        except Exception as e:
+            log.error(f"Error in _get_token: {str(e)}")
+            return None
             
-            log.error(f"Looking for user by oauth_sub: {provider_sub}, found: {user is not None}")
+    async def _get_user_data(self, token, provider_name):
```

Ruff (0.8.2)

Missing return type annotation for private function `_get_user_data`

(ANN202)